src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AmendReferralService.kt

```diff
@@ -0,0 +1,43 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.AmendComplexityLevelDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Changelog
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ChangelogRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ReferralRepository
+import java.time.OffsetDateTime
+import java.util.*
+import javax.transaction.Transactional
+
+@Service
+@Transactional
+class AmendReferralService(
+  val changelogRepository: ChangelogRepository,
+  val referralRepository: ReferralRepository
+) {
+
+  fun updateComplexityLevel(referral: Referral, update: AmendComplexityLevelDTO, actor: AuthUser): AmendComplexityLevelDTO? {
+    val existingDetails = referralRepository.findById(referral.id).get()
```

Comment:
I may be reading it wrong, but havn't you already got the referral details being passed into the method?